source/common/mongo/proxy.cc

```diff
@@ -50,12 +56,16 @@ ProxyFilter::ProxyFilter(const std::string& stat_prefix, Stats::Scope& scope,
 ProxyFilter::~ProxyFilter() {}
```

Comment:
If there is a disconnect, the timer will still exist and will eventually callback when the cnx doesn't exist.

Reply:
when ProxyFilter is destroyed, unique ptr (https://github.com/lyft/envoy/pull/1519/files#diff-8fff5f501b19d4f27e10798d8900a387R182) which holds the timer will deallocate memory used for it.

Reply:
See all other filters. Do to deferred deletion you must destroy timer at disconnect time.

Reply:
Add assert for the delay_timer ex https://github.com/lyft/envoy/blob/master/source/common/http/filter/fault_filter.cc#L76

Reply:
sounds good